tests/functional/sources/test_source_fresher_state.py

```diff
@@ -0,0 +1,727 @@
+import os
+import json
+import shutil
+import pytest
+from datetime import datetime, timedelta
+
+from dbt.exceptions import InternalException
+
+
+import dbt.version
+from tests.functional.sources.common_source_setup import BaseSourcesTest
+
+from tests.functional.sources.fixtures import (
+    error_models__schema_yml,
+    models__newly_added_model_sql,
+    models__newly_added_error_model_sql,
+)
+
+
+# TODO: We may create utility classes to handle reusable fixtures.
+def copy_to_previous_state():
+    shutil.copyfile("target/manifest.json", "previous_state/manifest.json")
+    shutil.copyfile("target/run_results.json", "previous_state/run_results.json")
```

Comment:
Not sure if Windows will be happy about this. We do have a `copy_file` utility [here](https://github.com/dbt-labs/dbt-core/blob/83612a98b733a02b78c55648123456650ee43a04/core/dbt/tests/util.py#L97), but it looks like it's doing pretty much the same thing.